mmv1/third_party/terraform/services/compute/resource_compute_instance_template.go.erb

```diff
@@ -1341,10 +1345,13 @@ func flattenDisk(disk *compute.AttachedDisk, configDisk map[string]any, defaultP
 		diskMap["disk_type"] = disk.InitializeParams.DiskType
 		diskMap["disk_name"] = disk.InitializeParams.DiskName
 		diskMap["labels"] = disk.InitializeParams.Labels
-		// The API does not return a disk size value for scratch disks. They can only be one size,
-		// so we can assume that size here.
-		if disk.InitializeParams.DiskSizeGb == 0 && disk.Type == "SCRATCH" {
+		// The API does not return a disk size value for scratch disks. They are largely only one size,
+		// so we can assume that size here. Prefer disk.DiskSizeGb over the deprecated
+		// disk.InitializeParams.DiskSizeGb.
+		if disk.DiskSizeGb == 0 && disk.InitializeParams.DiskSizeGb == 0 && disk.Type == "SCRATCH" {
```

Comment:
Possible logic bug, past and present: If the disk is not of type SCRATCH, we move past the if into the else-if and else blocks. I can put a guard here wrapping this entire section checking for SCRATCH, but it more closely emulates the existing code to not do so.

Reply:
This looks correct as-is to me. If the disk type is not `SCRATCH` and both `DiskSizeGb` fields are 0 we can set `disk_size_gb` to 0.